compiler/front_end/dependency_checker.py

```diff
@@ -207,9 +224,10 @@ def strong_connect(node):
 
 def _find_object_dependency_cycles(ir):
   """Finds dependency cycles in types in the ir."""
-  dependencies = _find_dependencies(ir)
+  dependencies, errors = _find_dependencies(ir)
+  if errors:
+    return errors
   cycles = _find_cycles(dict(dependencies))
-  errors = []
```

Comment:
I think this means that `_find_object_dependency_cycles()` can now return `None` instead of `[]`. That should cause an exception later on when this append is attempted:
`errors + _find_object_dependency_cycles(ir)`
If _find_object_dependency_cycles(ir) returns a list then this becomes
`None + []`
which I suspect doesn't work.

Reply:
It was actually assigned a couple lines above.  I have refactored slightly to avoid reusing the variable and make things a bit clearer.